removed references to Che from devfile docs

jc-berger opened the pull request:
For issue [#76](https://github.com/devfile/api/issues/76). 

I simply removed references to Che in the devfile docs. If the content no longer makes sense without the Che reference, more revision will be necessary.

sleshchenko commented on docs/modules/user-guide/partials/proc_adding-plugin-component-to-a-devfile.adoc:
```diff
@@ -23,7 +23,7 @@ This section describes how to add a `plugin` component to a devfile.
    - name: exec-plugin
      plugin:
-      id: eclipse/che-machine-exec-plugin/0.0.1
+      id: eclipse/machine-exec-plugin/0.0.1
```
maybe it would be more consistent not to reference eclipse at all. here below
```suggestion
      id: organization/sample-plugins/0.0.1
```

sleshchenko commented on docs/modules/user-guide/partials/proc_limiting-resources-usage.adoc:
```diff
@@ -30,7 +30,7 @@ This limit will be applied to every container of the given component.
 For `plugin` components, RAM limits can be described in the plug-in descriptor file, typically named `meta.yaml`.
 +
-If none of them are specified, system-wide defaults will be applied (see description of `pass:[CHE_WORKSPACE_SIDECAR_DEFAULT__MEMORY__LIMIT__MB]` system property).
+If none of them are specified, system-wide defaults will be applied (see description of `pass:[WORKSPACE_SIDECAR_DEFAULT__MEMORY__LIMIT__MB]` system property).
```
that's not actual for devfile I believe. Maybe it's better to just remove it

jc-berger commented:
@sleshchenko, thanks for the review! I've addressed your comments and applied the feedback to all the docs. The new, squashed commit is up now. Thanks!

elsony commented on docs/modules/user-guide/partials/proc_adding-container-component-to-a-devfile.adoc:
```diff
@@ -83,7 +83,7 @@ components:
 ====
 +
-The sources is mounted on a location stored in the `CHE_PROJECTS_ROOT` environment variable that is made available in the running container of the image. This location defaults to `/projects`. If `sourceMapping` is defined in the container, it overrides the 'CHE_PROJECT_ROOT' value if present and mounts the source to the path defined by `sourceMapping`.
+The sources is mounted on a location stored in the `ROJECTS_ROOT` environment variable that is made available in the running container of the image. This location defaults to `/projects`. If `sourceMapping` is defined in the container, it overrides the 'PROJECT_ROOT' value if present and mounts the source to the path defined by `sourceMapping`.
```
Typo `PROJECTS_ROOT`

elsony commented on docs/modules/user-guide/partials/proc_limiting-resources-usage.adoc:
```diff
@@ -86,7 +84,7 @@ This limit will be applied to every container of the given component.
 For the `plugin` components, CPU limits can be described in the plug-in descriptor file, typically named `meta.yaml`.
 +
-If none of them are specified, system-wide defaults are applied (see description of `pass:[CHE_WORKSPACE_SIDECAR_DEFAULT__CPU__LIMIT__CORES]` system property).
+If none of them are specified, system-wide defaults are applied (see description of `pass:[WORKSPACE_SIDECAR_DEFAULT__CPU__LIMIT__CORES]` system property).
```
I am not sure if this system property exists and how plugin behaves when no default is set in this case. @davidfestal @l0rd  We are trying to make the devfile spec Che independent. Do you have any suggestions on how to approach this?

l0rd replied:
I think that this line should be removed. If the devfile doesn't specify components requests/limits, the client (`odo` or `che` for example) will decide if and how to infer a default value. And Kubernetes will use `LimitRanges` if those are specified in the target namespaces. From a devfile spec we should say that, if they are not specified, the values are undetermined: they may (or may not) be inferred from the application that consumes the devfile or from Kubernetes.

elsony commented on docs/modules/user-guide/partials/proc_limiting-resources-usage.adoc:
```diff
@@ -115,4 +113,4 @@ This limit will be applied to every container of the given component.
 For the `pugin` component type, CPU requests can be described in the plug-in descriptor file, typically named `meta.yaml`.
 +
-If none of them are specified, system-wide defaults are applied (see description of `pass:[CHE_WORKSPACE_SIDECAR_DEFAULT__CPU__REQUEST__CORES]` system property).
+If none of them are specified, system-wide defaults are applied (see description of `pass:[WORKSPACE_SIDECAR_DEFAULT__CPU__REQUEST__CORES]` system property).
```
I am not sure if this system property exists and how plugin behaves when no default is set in this case. @davidfestal @l0rd  We are trying to make the devfile spec Che independent. Do you have any suggestions on how to approach this?

boczkowska commented:
LGTM for the lang perspective.

jc-berger commented:
@elsony and/or @l0rd, new, squashed commit is up. I've addressed the typo error and removed the suggested sentences. 

Let me know if either of you thinks it's ready to be merged. Thanks!

elsony commented on docs/modules/user-guide/partials/proc_limiting-resources-usage.adoc:
```diff
@@ -86,8 +84,6 @@ This limit will be applied to every container of the given component.
 +
 For the `plugin` components, CPU limits can be described in the plug-in descriptor file, typically named `meta.yaml`.
 +
-If none of them are specified, system-wide defaults are applied (see description of `pass:[CHE_WORKSPACE_SIDECAR_DEFAULT__CPU__LIMIT__CORES]` system property).
```
Based on Mario's previous comment, I think he mentioned that we need to add "From a devfile spec we should say that, if they are not specified, the values are undetermined: they may (or may not) be inferred from the application that consumes the devfile or from Kubernetes." after removing the existing line.

jc-berger replied:
Thanks, Elson, I overlooked that! I'll address these comments shortly.

elsony commented on docs/modules/user-guide/partials/proc_limiting-resources-usage.adoc:
```diff
@@ -59,7 +57,7 @@ This limit will be applied to every container of the given component.
 For `plugin` components, RAM requests can be described in the plug-in descriptor file, typically named `meta.yaml`.
 +
-If none of them are specified, system-wide defaults are applied (see description of `pass:[CHE_WORKSPACE_SIDECAR_DEFAULT__MEMORY__REQUEST__MB]` system property).
+If none of them are specified, system-wide defaults are applied (see description of `pass:[WORKSPACE_SIDECAR_DEFAULT__MEMORY__REQUEST__MB]` system property).
```
Same as the other case, we should remove this and add that "if they are not specified, the values are undetermined: they may (or may not) be inferred from the application that consumes the devfile or from Kubernetes." (reword if needed)

elsony commented on docs/modules/user-guide/partials/proc_limiting-resources-usage.adoc:
```diff
@@ -115,4 +111,3 @@ This limit will be applied to every container of the given component.
 +
 For the `pugin` component type, CPU requests can be described in the plug-in descriptor file, typically named `meta.yaml`.
 +
-If none of them are specified, system-wide defaults are applied (see description of `pass:[CHE_WORKSPACE_SIDECAR_DEFAULT__CPU__REQUEST__CORES]` system property).
```
Same as the other case, we should remove this and add that "if they are not specified, the values are undetermined: they may (or may not) be inferred from the application that consumes the devfile or from Kubernetes." (reword if needed)

elsony commented on docs/modules/user-guide/partials/proc_limiting-resources-usage.adoc:
```diff
@@ -115,4 +111,3 @@ This limit will be applied to every container of the given component.
 +
 For the `pugin` component type, CPU requests can be described in the plug-in descriptor file, typically named `meta.yaml`.
```
typo `plugin`

jc-berger commented:
Thanks, @elsony, new, squashed commit is up!

elsony reviewed (approved):
/lgtm

rkratky commented on docs/modules/user-guide/partials/proc_adding-commands-to-a-devfile.adoc:
```diff
@@ -58,29 +58,29 @@ components:
       component: go-cli
       commandLine: “go get -d && go run main.go”
-      workingDir: “${CHE_PROJECTS_ROOT}/src/github.com/acme/my-go-project”
+      workingDir: “${PROJECTS_ROOT}/src/github.com/acme/my-go-project”
```
Use ASCII characters only. There quotation marks were typographical, which I'm quite sure would cause an error.

```suggestion
      workingDir: "${PROJECTS_ROOT}/src/github.com/acme/my-go-project"
```

rkratky commented on docs/modules/user-guide/partials/proc_adding-a-name-to-a-devfile.adoc:
```diff
@@ -22,15 +22,15 @@ metadata:
+. To specify a prefix for automatically generated workspace names, define the `generateName` attribute and don't define the `name` attribute. The workspace name will be in the `<generateName>YYYYY` format (for example, `devfile-sample-2y7kp`). `Y` is random `[a-z0-9]` character.
 +
 .Adding a generated name to a devfile
 ====
```
No need for this extra block. Remove these blocks everywhere (I only suggested the change in a handful of examples, but they are present in many places).
```suggestion
```

rkratky commented on docs/modules/user-guide/partials/proc_adding-a-name-to-a-devfile.adoc:
```diff
@@ -22,15 +22,15 @@ metadata:
-  generateName: che-
+  generateName: devfile-sample-
 ----
 ====
```
```suggestion
```

rkratky commented on docs/modules/user-guide/partials/proc_adding-a-name-to-a-devfile.adoc:
```diff
@@ -10,7 +10,7 @@ Adding a name to a devfile is mandatory. Both `name` and `generateName` are opti
+. To specify a static name for the workspace, define the `name` attribute.
 +
 .Adding a static name to a devfile
 ====
```
```suggestion
```

rkratky commented on docs/modules/user-guide/partials/proc_adding-a-name-to-a-devfile.adoc:
```diff
@@ -22,15 +22,15 @@ metadata:
 ----
 ====
```
```suggestion
```

rkratky commented on docs/modules/user-guide/partials/proc_adding-commands-to-a-devfile.adoc:
```diff
@@ -58,29 +58,29 @@ components:
-If the editor in the workspace supports it, the devfile can specify additional configuration in the editor-specific format. This is dependent on the integration code present in the workspace editor itself and so is not a generic mechanism. However, the default Che-Theia editor within {che-prod} is equipped to understand the `tasks.json` and `launch.json` files provided in the devfile.
+If the editor in the workspace supports it, the devfile can specify additional configuration in the editor-specific format. This is dependent on the integration code present in the workspace editor itself and so is not a generic mechanism.
 +
 ====
```
```suggestion
```

rkratky commented on docs/modules/user-guide/partials/proc_adding-commands-to-a-devfile.adoc:
```diff
@@ -111,7 +111,7 @@ commands:
 
-This example shows association of a `tasks.json` file with a devfile. Notice the `vscode-task` type that instructs the Che-Theia editor to interpret this command as a tasks definition and `referenceContent` attribute that contains the contents of the file itself. You can also save this file separately from the devfile and use `reference` attribute to specify a relative or absolute URL to it.
+This example shows association of a `tasks.json` file with a devfile. Notice the `vscode-task` type that instructs the editor to interpret this command as a tasks definition and `referenceContent` attribute that contains the contents of the file itself. You can also save this file separately from the devfile and use `reference` attribute to specify a relative or absolute URL to it.
 ====
```
```suggestion
```

rkratky commented on docs/modules/user-guide/partials/proc_adding-container-component-to-a-devfile.adoc:
```diff
@@ -83,7 +83,7 @@ components:
 ====
 +
-The sources is mounted on a location stored in the `CHE_PROJECTS_ROOT` environment variable that is made available in the running container of the image. This location defaults to `/projects`. If `sourceMapping` is defined in the container, it overrides the 'CHE_PROJECT_ROOT' value if present and mounts the source to the path defined by `sourceMapping`.
+The sources is mounted on a location stored in the `PROJECTS_ROOT` environment variable that is made available in the running container of the image. This location defaults to `/projects`. If `sourceMapping` is defined in the container, it overrides the 'PROJECT_ROOT' value if present and mounts the source to the path defined by `sourceMapping`.
```
Remember to use fixed width (backticks) for variable names.

```suggestion
The sources is mounted on a location stored in the `PROJECTS_ROOT` environment variable that is made available in the running container of the image. This location defaults to `/projects`. If `sourceMapping` is defined in the container, it overrides the `PROJECT_ROOT` value if present and mounts the source to the path defined by `sourceMapping`.
```

rkratky reviewed (changes requested):
@jc-berger, thanks. Some fixes suggested.
